### PR TITLE
Ci/multiple jobs pipeline

### PR DESCRIPTION
**Pull request by jonathangoulding**
Break CI Pipeline into multiple jobs

Add matrix to establish any breaks from upgrading from node v12 to v14 (can extend to v16 in the matrix)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,14 +3,17 @@ name: CI
 on:
   push:
     branches:
-      - develop
+      - CI/multiple-jobs-pipeline
```
Comment (jonathangoulding) [inline .github/workflows/ci.yml:6]: Put back as develop. 

propose CI on all branches - inc feature branches

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
   pull_request:
     types: [ opened, synchronize, reopened ]
 
 jobs:
-  build:
-    # You must use a Linux environment when using service containers or container jobs
+  base-node-pipeline: 
+    uses: DEFRA/water-abstraction-orchestration/.github/workflows/base-node-pipeline.yml@develop
```
Comment (timtamimi) [inline .github/workflows/ci.yml:13]: This is clever. 

Presumably we can do the same across the other repos (UI/Returns/Permits/Import) etc? :D

Comment (jonathangoulding) [reply .github/workflows/ci.yml:13]: All should / can share the common pipeline. 

Best to discuss as a team on the common jobs and what node versions to test

Comment (jonathangoulding) [reply .github/workflows/ci.yml:13]: test setup looks a little different per repo so keep the test ci as a job in each repo.

**Comment by jonathangoulding** [comment]
Currently allowing audit to ` continue-on-error: true`

Job added to highlight potential vulnerabilities and issues with packages

**Pull request closed without merging**